vm/devices/storage/disk_nvme/nvme_driver/src/queue_pair.rs

```diff
@@ -239,13 +261,26 @@ impl QueuePair {
             }
         });
 
-        // Page allocator uses remaining part of the buffer for dynamic allocation.
-        const _: () = assert!(
-            QueuePair::PER_QUEUE_PAGES * PAGE_SIZE >= 128 * 1024 + PAGE_SIZE,
-            "not enough room for an ATAPI IO plus a PRP list"
-        );
-        let alloc: PageAllocator =
-            PageAllocator::new(mem.subblock(data_offset, QueuePair::PER_QUEUE_PAGES * PAGE_SIZE));
+        // Page allocator uses remaining part of the buffer for dynamic
+        // allocation. The length of the page allocator depends on if bounce
+        // buffering / double buffering is needed.
+        let alloc_len = if bounce_buffer {
+            const SIZE: usize = QueuePair::PER_QUEUE_PAGES_BOUNCE_BUFFER * PAGE_SIZE;
+            const _: () = assert!(
```

Comment:
Can we package this up as a local const fn? E.g.,

```rust
const fn pages_to_size(pages: usize) -> usize {
    let size = pages * PAGE_SIZE;
    assert!(size >= ...);
    size
}
// Note: using const {} to ensure the internal assertion checks are at compile time
let alloc_len = if bounce_buffer {
    const { pages_to_size(QueuePair::PER_QUEUE_PAGES_BOUNCE_BUFFER) }
} else {
    const { pages_to_size(QueuePair::PER_QUEUE_PAGES_NO_BOUNCE_BUFFER) }
};
```

Reply:
neat, can do